Blaze/ByteString/Builder/HTTP.hs

```diff
@@ -182,9 +182,9 @@ chunkedTransferEncoding innerBuilder =
             minimalChunkSize  = 1
 
             -- overhead computation
-            maxBeforeBufferOverhead = sizeOf (undefined :: Int) + 2 -- max chunk size and CRLF after header
-            maxAfterBufferOverhead  = 2 +                           -- CRLF after data
-                                      sizeOf (undefined :: Int) + 2 -- max bytestring size, CRLF after header
+            maxBeforeBufferOverhead = sizeOf (undefined :: Word32) * 2 + 2 -- max chunk size and CRLF after header
+            maxAfterBufferOverhead  = 2 +                                  -- CRLF after data
+                                      sizeOf (undefined :: Word32) * 2 + 2 -- max bytestring size, CRLF after header
```

Comment:
Thanks for your contribution.

`sizeOf (Int)` gives the size of machine words.  Why does it not work correctly on 32 bit systems?  It should give the same value there as `sizeOf (Word32)`, namely 4.
Looking from afar, I'd say this change only changes things on non-32 bit systems.  E.g. on my current machine, `sizeOf (Int) = 8` and `sizeOf (Word32) = 4`.
Do you think that making these values machine-dependent was a misconception from the beginning?

The current code does not give an argumentation why these values should be machine-dependent, this is unfortunately some technical dept.
If you have an argumentation why your change makes things more correct, please add it in the form of a comment.

Reply:
The protocol part of HTTP chunked encoding uses ASCII and encodes all offsets in textual forms. Thus, these overhead values depend on the maximal size of a Word32 encoded in hexadecimal notation which is 8 characters (bsb-http-chunked just hardcodes this value). This encoding (in the worst cases) uses twice the bytes as the machine representation (at least for platforms I know of), hence I've chosen `sizeOf (_::Word32) * 2`. This value just so happens to be the same as `sizeOf (_::Int)` on 64 bit platforms.

Probably, just using hardcoded 8 is the most portable since this value doesn't depend on machine sizes *at all*.

Reply:
We could also use `word32HexLength`, but I'm not sure whether GHC is able to statically compute that.

Reply:
Ah thanks for the explanations!
Could you put these as Haskell comments adjacent to the code?
I like the idea to define `word32HexLength = sizeOf (_::Word32) * 2` to give a name to this quantity.

Does this code need update as well? https://github.com/blaze-builder/blaze-builder/blob/fcdc60f48875f0f74f1a7ed378b99736f159efd7/Blaze/ByteString/Builder.hs#L156-L161

Reply:
I was referring to

https://github.com/blaze-builder/blaze-builder/blob/fcdc60f48875f0f74f1a7ed378b99736f159efd7/Blaze/ByteString/Builder/HTTP.hs#L96-L99

so we could have `work32HexLength (maxBound :: Word32)`.



> I like the idea to define `word32HexLength = sizeOf (_::Word32) * 2` to give a name to this quantity.

We could also do that, yes. It seems to me that just having e.g. `maxChunkSizeLength = 8` makes the most sense since the value is not machine size dependent at all.

Reply:
> It seems to me that just having e.g. `maxChunkSizeLength = 8`

Fine with me as well, as long as there is documentation of this magic number.

Reply:
Done. LMK what you think.

Reply:
Looks good!